
* Работа для агенства "Digital Scouting"
* Данные по зарплатам игроков примерно 30 ведущих футбольных лиг.

# Содержание

1. [Подготовим данные и сделаем первые выводы](#view1)
2. [Зарплатный рейтинг по возврастам](#view5)
  - 2.1 [Итоговая таблица по возрастам](#age1)
3. [Зарплатный рейтинг по позициям](#view4)
  - 3.1 [Итоговая таблица по позициям](#poz1)
4. [Национальный рейтинг](#naz1)
  - 4.1 [Национальный рейтинг итоговая таблица](#nazf)
5. [Национальный рейтинг по зп](#naz2)
  - 5.1 [Национальный рейтинг по игрокам итоговая таблица](#nazf2)
6. [Рейтинг клубов по ЗП ведомости](#view2) 

 - 6.1 [Итог таблица перцентили](#final1)
 - 6.2 [Итог таблица по игрокам](#final2)
 
7. [Рейтинг лиг по зп ведомости](#view3)
  - 7.1 [Итоговая таблица по рейтингу лиг](#zp1)


8. [Вывод](#vivod1)

## 1.Подготовим данные и сделаем первые выводы<a name="view1"></a>

In [146]:
import pandas as pd
import json
from IPython.display import display
import sklearn
from sklearn.cluster import KMeans
import numpy as np
import math
import matplotlib.pyplot as plt
import os
#увеличим вывод строк
pd.options.display.max_rows = 1000

pd.set_option('display.max_colwidth', None)
pd.set_option('display.float_format', '{:.0f}'.format)

In [147]:
data = pd.read_csv('nazrat.csv',sep = ",",low_memory=False)
data = data.replace("\'", '\"')

In [148]:
display(data.head(12))
display(data.shape)

player  net_pw         net_py position  age      country  \
0        Kylian Mbappé  977432  5082648000001        F   23       France   
1         Lionel Messi  684203  3557853600002        F   34    Argentina   
2               Neymar  606008  3151241800003        F   29       Brazil   
3    Cristiano Ronaldo  579644  3014150700004        F   36     Portugal   
4          Gareth Bale  319034  1658976300005        F   32        Wales   
5          Eden Hazard  293230  1524794400006        F   31      Belgium   
6         Sergio Ramos  293230  1524794400007        D   35        Spain   
7      Frenkie de Jong  273681  1423141400008        M   24  Netherlands   
8      Kevin De Bruyne  265245  1379275400009        F   30      Belgium   
9         David de Gea  248783  1293673500010        K   31        Spain   
10  Robert Lewandowski  238298  1239149600011        F   33       Poland   
11        Jadon Sancho  232090  1206866000012        F   21      England   

                 club          league  
0                 PSG         Ligue 1  
1                 PSG         Ligue 1  
2                 PSG         Ligue 1  
3   Manchester United  Premier League  
4         Real Madrid         La Liga  
5         Real Madrid         La Liga  
6                 PSG         Ligue 1  
7           Barcelona         La Liga  
8     Manchester City  Premier League  
9   Manchester United  Premier League  
10      Bayern Munich    1.Bundesliga  
11  Manchester United  Premier League

(14291, 8)

In [149]:
# с помощью лямб функции поменяем тип данных с object  на  float для того что бы произвести расчеты
#data['net_py'] = data['net_py'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')
#data['net_pw'] = data['net_pw'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')


## 5. Национальный рейтинг по зп <a name="naz2"></a>

Национальность	Наивысшая зарплата	Зарплата 10 игрока	Зарплата 30 игрока

In [150]:
zp = data.groupby(['country']).agg({'net_py':'max'})
zp.sort_values(by='net_py',ascending = False,inplace=True)
display(zp.shape)
zp.head(1)

(153, 1)

net_py
country               
France   5082648000001

In [151]:
filter_list=[10,30]

In [152]:
#data_coutry = data.copy()
#data_coutry['rank'] = data_coutry.groupby('country')['net_py'].rank(method='dense',ascending=False)
#data_coutry_filtered = data_coutry.query("rank == @filter_list")
#data_coutry_filtered.sort_values(by = 'rank', ascending = False)
#display(data_coutry_filtered.shape)
#data_coutry_filtered.head(305)


data['rank'] = data.groupby('country')['net_py'].rank(method='dense',ascending=False)
data_coutry_filtered = data.query("rank == @filter_list")
data_coutry_filtered.sort_values(by = 'rank', ascending = False)
display(data_coutry_filtered.shape)
data_coutry_filtered.head(3)

(143, 9)

player  net_pw        net_py position  age country  \
38   Kingsley Coman  176133  915893200039        F   25  France   
81    Cesc Fàbregas  122375  636347500082        M   34   Spain   
103     Fernandinho   99467  517228300104        M   36  Brazil   

                club          league  rank  
38     Bayern Munich    1.Bundesliga    10  
81            Monaco         Ligue 1    10  
103  Manchester City  Premier League    10

In [153]:
data_pivot = pd.pivot_table(data_coutry_filtered,
               index='country',
               values="net_py",
               columns="rank"
               )

display(data_pivot.shape)
data_pivot

(89, 2)

rank                                       10           30
country                                                   
Albania                           47776902718   5082609680
Algeria                          121983601008  16264505907
Angola                            25413204477          NaN
Argentina                        325289500251 152479400780
Australia                         19314105303   3617011336
Austria                          124016600992  45000002848
Belgium                          309854700267  98603401302
Benin                              3049611649          NaN
Bosnia-Herzegovina                63024802015  20330605131
Brazil                           517228300104 276096200318
Bulgaria                           4066110763          NaN
Burkina Faso                       6099209255          NaN
Cameroon                          51843002438  14231406383
Canada                            26524504288   6600009045
Cape Verde                         3049611456          NaN
Chile                             78272801632  20496205088
Colombia                         142314100840  57871702237
Congo                              6099209174          NaN
Costa Rica                        19314105389          NaN
Cote d'Ivoire                    152479400748  50826502513
Croatia                          221603500457  60283002121
Curacao                            5082609741          NaN
Cyprus                                  14198          NaN
Czech Republic                    86807501448   7115708723
Democratic Republic of the Congo  57942202220   9148807860
Denmark                          169760400660  68722601880
Ecuador                           56000002276  25920004397
Egypt                             17281005707          NaN
England                          517228300110 309854700272
Finland                           32528903734   2033112767
France                           915893200039 341201900239
Gabon                              7115708812          NaN
Georgia                           10165307409          NaN
Germany                          517228300111 234818300416
Ghana                             95553801329  30495903877
Greece                            41677703049  19314105383
Guadeloupe                         5082609829          NaN
Guinea                            19314105364          NaN
Guinea-Bissau                      6099209261          NaN
Haiti                              1016513693          NaN
Honduras                           5082609873          NaN
Hungary                           30495903867          NaN
Iceland                           26429804332          NaN
Iran                               9000008001          NaN
Ireland                           86807501472  32552803723
Israel                            13214906577          NaN
Italy                            406611800183 203305900521
Jamaica                           51843402422   6028309436
Japan                             48793402641  15247906092
Kosovo                            53876102362   5082609957
Luxembourg                         4003450794          NaN
Mali                              37500003344  12198406766
Martinique                         9148807827          NaN
Mexico                           142000000844  64000002005
Montenegro                        10165307447          NaN
Morocco                           96452801325  39644703185
Netherlands                      259217000358 139856600863
New Zealand                        6028309402          NaN
Nigeria                           95247201343  46760402748
North Macedonia                    4066110786          NaN
Northern Ireland                  36169803413   8439608065
Norway                           111818301109  35578503493
Panama                             5000010064          NaN
Paraguay                          91800001371  23000004789
Peru                              43000002959          NaN
Poland                           142314100832  

## 5.1. Национальный рейтинг по игрокам итоговая табица<a name="nazf2"></a>

In [154]:
rating_naz = data_pivot.merge(zp ,left_on='country' ,right_on = 'country')
display(rating_naz.shape)
rating_naz = rating_naz.sort_values(by='net_py',ascending = False)
rating_naz = rating_naz.fillna(0)
rating_naz 

(89, 3)

10.0         30.0         net_py
country                                                                  
France                           915893200039 341201900239  5082648000001
Argentina                        325289500251 152479400780  3557853600002
Brazil                           517228300104 276096200318  3151241800003
Portugal                         318173800261 139999980861  3014150700004
Wales                            103686801186  25318904485  1658976300005
Spain                            636347500082 363917600211  1524794400007
Belgium                          309854700267  98603401302  1524794400006
Netherlands                      259217000358 139856600863  1423141400008
Poland                           142314100832  45743802819  1239149600011
England                          517228300110 309854700272  1206866000012
Germany                          517228300111 234818300416  1131397400015
Austria                          124016600992  45000002848  1097852000018
Italy                            406611800183 203305900521  1016529600029
Slovenia                          66000001955  15247906184  1014496500031
Croatia                          221603500457  60283002121   960620500037
Uruguay                          195173700547  65000001983   862047100042
Gabon                              7115708812            0   862047100041
Morocco                           96452801325  39644703185   813223700045
Denmark                          169760400660  68722601880   762397200051
Chile                             78272801632  20496205088   711570700061
Sweden                            86807501453  33545503622   711570700059
Serbia                           162644700694  46760402762   711570700057
Ghana                             95553801329  30495903877   689637700069
Egypt                             17281005707            0   689637700067
Costa Rica                        19314105389            0   670909500073
South Korea                       29479404055            0   661907500074
Senegal                          183991900582  41677703037   609917800090
Canada                            26524504288   6600009045   605851600092
United States                    102600001226  52500002390   517228300109
Colombia                         142314100840  57871702237   508264800121
Turkey                           144347200821  78272801643   508264800118
Bosnia-Herzegovina                63024802015  20330605131   508264800117
Cote d'Ivoire                    152479400748  50826502513   482264100128
Nigeria                           95247201343  46760402748   457438300145
Mexico                           142000000844  64000002005   457438300144
Norway                           111818301109  35578503493   425925900156
Algeria                          121983601008  16264505907   413541500170
Guinea                            19314105364            0   413541500163
Greece                            41677703049  19314105383   406611800184
Ukraine                           71157101773            0   396662200189
Scotland                         115200001072  45743802822   379783000197
Iceland                           26429804332            0   344818800237
Switzerland                      137445300884  48793402665   344818800235
Cameroon                          51843002438  14231406383   344818800234
Russia                           176876200613 106735601164   327322500248
Slovakia                          58958702190        14239   304958900285
Northern Ireland                  36169803413   8439608065   276096200322
New Zealand                        6028309402            0   276096200316
Burkina Faso                       6099209255            0   268862200337
Democratic Republic of the Congo  57942202220   9148807860   268363800339
Japan                             48793402641  15247906092   259217000365
Suriname                           2033112585            0   254132400381
Ecuador                           56000002276

In [170]:
en = data.query('country == "England"')
en.sort_values(by='rank',ascending = True).head(11)

player  net_pw         net_py position  age  country  \
11             Jadon Sancho  232090  1206866000012        F   21  England   
25          Raheem Sterling  198934  1034456500026        F   27  England   
24            Jack Grealish  198934  1034456500025        F   26  England   
42              John Stones  165778   862047100043        D   27  England   
67               Harry Kane  132623   689637700068        F   28  England   
65          Marcus Rashford  132623   689637700066        F   24  England   
78            Harry Maguire  125899   654673500079        D   28  England   
77             Ben Chilwell  125899   654673500078        D   25  England   
84   Trent Alexander-Arnold  119407   620915100085        D   23  England   
109               Luke Shaw   99467   517228300110        D   26  England   
126         Kieran Trippier   95757   497937700127        D   31  England   

                  club          league  rank  
11   Manchester United  Premier League     1  
25     Manchester City  Premier League     2  
24     Manchester City  Premier League     3  
42     Manchester City  Premier League     4  
67           Tottenham  Premier League     5  
65   Manchester United  Premier League     6  
78   Manchester United  Premier League     7  
77             Chelsea  Premier League     8  
84           Liverpool  Premier League     9  
109  Manchester United  Premier League    10  
126          Newcastle  Premier League    11

In [156]:
ger = data.query('country == "Germany"')
ger

player  net_pw         net_py position  age  \
14                      Manuel Neuer  217576  1131397400015        K   36   
16                     Thomas Müller  212494  1104967700017        F   32   
18                        Leroy Sané  207216  1077521400019        F   26   
21                        Toni Kroos  205847  1070405700022        M   32   
23                    Joshua Kimmich  202133  1051091600024        M   26   
35                     Leon Goretzka  186494   969769200036        M   26   
37                       Timo Werner  180385   938003700038        F   25   
92                        Marco Reus  113969   592636800093        F   32   
100                     Mats Hummels  103608   538760700101        D   33   
110                      Kai Havertz   99467   517228300111        F   22   
130                   Ilkay Gündogan   92743   482264100131        M   31   
134                         Emre Can   91097   473702800135        M   28   
150                  Alexander Nübel   85037   442190400151        K   25   
151            Marc-André ter Stegen   84450   439140800152        K   29   
153                     Serge Gnabry   82886   431008600154        F   26   
187                   Julian Draxler   77413   402545700188        F   28   
200                      Niklas Süle   72525   377132500201        D   26   
227                       Bernd Leno   66311   344818800228        K   29   
232                  Antonio Rüdiger   66311   344818800233        D   28   
239                       Mesut Özil   64902   337487800240        F   33   
259                    Kevin Volland   61187   318173800260        F   29   
299                Maximilian Arnold   57082   296826600300        M   27   
326                     Thilo Kehrer   52977   275479500327        D   25   
334                    Jamal Musiala   51804   269380300335        F   18   
369                     Robin Gosens   48872   254132400370        M   27   
370                     Nadiem Amiri   48872   254132400371        F   25   
382                      Diego Demme   48872   254132400383        M   30   
397                 Dominick Drexler   46721   242950600398        F   31   
414                   Sebastian Rudy   45548   236851400415        M   31   
415                   Jérôme Boateng   45157   234818300416        D   33   
466                    Julian Brandt   41443   215504300467        F   25   
467                   Mahmoud Dahoud   41443   215504300468        M   26   
503                    Simon Terodde   39684   206355500504        F   33   
506                        Max Kruse   39293   204322400507        F   33   
550                 Marcel Schmelzer   37338   194157200551        D   34   
551                 Yannick Gerhardt   37338   194157200552        M   27   
572                 Christoph Kramer   36360   189074500573        M   30   
606                     Jonas Hector   34210   177892700607        D   31   
607                      Marius Wolf   34210   177892700608        F   26   
619                   Daniel Ginczek   33233   172810000620        F   30   
620                   Kerem Demirbay   33233   172810000621        M   28   
639                      Pascal Groß   33156   172409400640        M   30   
657                   Oliver Baumann   32646   169760400658        K   31   
661                      Suat Serdar   32451   168743900662        M   24   
672                  Matthias Ginter   32060   166710900673        D   28   
704               Marcel Halstenberg   31082   161628200705        D   30   
707                   Robert Andrich   30887   160611700708        M   27   
713                       Robin Koch   30605   159147200714        D   25   
717                      Lars Stindl   30105   156545600718        F   33   
718                       Edgar Prib   30105   156545600719        M   32   
735                   Dominik Kaiser   29714   154512500736        M   33   
745                     Ju

In [160]:
es = data.query('country == "Spain"')
es

player  net_pw         net_py position  age country  \
6                  Sergio Ramos  293230  1524794400007        D   35   Spain   
9                  David de Gea  248783  1293673500010        K   31   Spain   
19              Sergio Busquets  206434  1073455300020        M   33   Spain   
26                   Jordi Alba  195486  1016529600027        D   32   Spain   
48                         Koke  150134   780694700049        M   30   Spain   
64                       Thiago  132623   689637700065        M   30   Spain   
69                  Saúl Ñíguez  131463   683609400070        M   27   Spain   
70                    Ansu Fati  130976   681074800071        F   19   Spain   
80                         Isco  123156   640413600081        F   29   Spain   
81                Cesc Fàbregas  122375   636347500082        M   34   Spain   
97                    Juan Mata  106191   552192400098        F   33   Spain   
102                 Dani Parejo  101653   528595400103        M   32   Spain   
105           César Azpilicueta   99467   517228300106        D   32   Spain   
111           Kepa Arrizabalaga   99467   517228300112        K   27   Spain   
114                Gerard Piqué   97743   508264800115        D   35   Spain   
123               Álvaro Morata   97743   508264800124        F   29   Spain   
133               Marco Asensio   92270   479802000134        F   26   Spain   
136                 Juan Bernat   90315   469636700137        D   28   Spain   
139                       Pedri   87969   457438300140        M   19   Spain   
149             Daniel Carvajal   86210   448289600150        D   30   Spain   
152               Ander Herrera   83864   436091200153        M   32   Spain   
156                  Pablo Marí   80736   419826700157        D   28   Spain   
157                       Rodri   80455   418364100158        M   25   Spain   
171             Aymeric Laporte   79527   413541500172        D   27   Spain   
175               Lucas Vázquez   79172   411694500176        D   30   Spain   
186                 Raúl García   77999   405595300187        F   35   Spain   
203             Nacho Fernández   72135   375099400204        D   32   Spain   
206               Gerard Moreno   70375   365950700207        F   29   Spain   
209                  Pau Torres   70375   365950700210        D   25   Spain   
210               Óliver Torres   69984   363917600211        M   27   Spain   
217            Álvaro Odriozola   68420   355785400218        D   26   Spain   
228               Marcos Alonso   66311   344818800229        D   31   Spain   
246                      Vitolo   63338   329355600247        F   32   Spain   
251             Héctor Bellerín   62556   325289500252        D   26   Spain   
272                        Suso   59037   306991900273        F   28   Spain   
276               Ferran Torres   58646   304958900277        F   21   Spain   
305              Iñaki Williams   56105   291744000306        F   27   Spain   
306               Pablo Sarabia   54932   285644800307        F   29   Spain   
314                   Bryan Gil   53095   276096200315        F   20   Spain   
327              Vicente Iborra   52781   274463000328        M   34   Spain   
340                 Diego Costa   51538   268000000341        F   32   Spain   
349              Vicente Guaita   51009   265245300350        K   35   Spain   
366                Paco Alcácer   48872   254132400367        F   28   Spain   
371                       Pedro   48872   254132400372        F   34   Spain   
375                Luis Alberto   48872   254132400376        M   29   Spain   
394              Iñigo Martínez   46917   243967100395        D   30   Spain   
417               Borja Mayoral   44962   233801800418        F   24   Spain   
418                 Raúl Albiol   44962   233801800419        D   36   Spain   
422               Pablo Fornals   44749   232692400423        F   25   Spain   
424                 Joan 

In [162]:
be = data.query('country == "Belgium"')
be

player  net_pw         net_py position  age  country  \
5                 Eden Hazard  293230  1524794400006        F   31  Belgium   
8             Kevin De Bruyne  265245  1379275400009        F   30  Belgium   
15              Romelu Lukaku  215628  1121264100016        F   28  Belgium   
94           Thibaut Courtois  110841   576372300095        K   29  Belgium   
135           Michy Batshuayi   90901   472686300136        F   28  Belgium   
142             Dries Mertens   87969   457438300143        F   34  Belgium   
154               Axel Witsel   82886   431008600155        M   33  Belgium   
161         Christian Benteke   79527   413541500162        F   31  Belgium   
205          Yannick Carrasco   70962   369000200206        F   28  Belgium   
266        Leander Dendoncker   59587   309854700267        M   26  Belgium   
463          Timothy Castagne   41734   217018900464        D   26  Belgium   
491              Divock Origi   39880   207373600492        F   26  Belgium   
549             Adnan Januzaj   37533   195173700550        F   26  Belgium   
584             Koen Casteels   35188   182975300585        K   29  Belgium   
603      Albert Sambi Lokonga   34547   179643400604        M   22  Belgium   
624            Simon Mignolet   33233   172810000625        K   33  Belgium   
770              Dennis Praet   29323   152479400771        M   27  Belgium   
802             Jason Denayer   28346   147396800803        D   26  Belgium   
804          Leandro Trossard   28287   147090600805        F   27  Belgium   
809            Jan Vertonghen   28150   146380300810        D   34  Belgium   
852            Thomas Meunier   26977   140281100853        D   30  Belgium   
877         Charly Musonda Jr   26432   137445300878        F   25  Belgium   
896            Dodi Lukébakio   26000   135198400897        F   24  Belgium   
952            Dedryck Boyata   24827   129099300953        D   31  Belgium   
1078          Youri Tielemans   22026   114537701079        M   24  Belgium   
1194       Christian Kabasele   19940   103686801195        D   30  Belgium   
1228         Radja Nainggolan   19549   101653001229        M   33  Belgium   
1232            Jordan Lukaku   19549   101653001233        D   27  Belgium   
1267      Alexis Saelemaekers   19549   101653001268        F   22  Belgium   
1301       Sebastiaan Bornauw   18962    98603401302        D   22  Belgium   
1382             Hans Vanaken   17594    91487701383        F   29  Belgium   
1475           Thorgan Hazard   16616    86405001476        F   28  Belgium   
1538            Dries Wouters   15834    82338901539        M   24  Belgium   
1673   Hendrik Van Crombrugge   14661    76239701674        K   28  Belgium   
1711        Birger Verstraete   14271    74206701712        M   27  Belgium   
1825          Dino Arslanagic   13489    70140501826        D   28  Belgium   
1836           Boli Bolingoli   13293    69124001837        D   26  Belgium   
2013                Sven Kums   12120    63024802014        M   33  Belgium   
2075              Maxime Busi   11729    60991802076        D   22  Belgium   
2115              Ruud Boffin   11729    60991802116        K   34  Belgium   
2162                Matz Sels   11534    59975202163        K   29  Belgium   
2213         Maxime Lestienne   11143    57942202214        F   29  Belgium   
2214             Jelle Vossen   11143    57942202215        F   32  Belgium   
2217            Landry Dimata   11143    57942202218        F   24  Belgium   
2280         Laurent Depoitre   10752    55909102281        F   33  Belgium   
2315             Renaud Emond   10556    54892602316        F   30  Belgium   
2325             Nacer Chadli   10556    54892602326        F   32  Belgium   
2402             Isaac Mbenza    9970    51843402403        F   25  Belgium   
2492         Anthony Limbombe    9774    50826502493        F   27  Belgium   
2698          Zakaria Bakkali    9188    47776902699        F   26  Belgi

## Вывод:
* По максимально зарплате в клубе лидеры Франция, Аргентина,Бразилия и Португалия, думаю можно без проблем называть этих игроков. Франция сильно в отрыве по игрокам из всех корзин. 

* Только у Бразили и Испании оказались 200е и 100е игроки у которых больши разброс по зп.

* Нули представлены здесь как замена пропущенным значениям, не все игроки  дотягивают до отметки 100е и 200е и ранк кидает их в другую корзину. Для того что бы проверить это оставил вычесления выше. 

* Получается, что несмотря на количество игроков зп могут быть сильно в разных диапозонах.

##  Найдем суммы зарплат 
(платежные ведомости) всех команд. 
* найти где граница между топ-клубами и просто сильными командами.

In [72]:
data_new = pd.read_csv('rank.csv',sep = ",",low_memory=False)
data_new = data_new.replace("\'", '\"')

In [73]:
# с помощью лямб функции поменяем тип данных с object  на  float для того что бы произвести расчеты
#data_new['net_py'] = data_new['net_py'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')
#data_new['net_pw'] = data_new['net_pw'].apply(lambda x: x.replace('$', '').replace(',', '')).astype('float')



In [74]:
# в расчет берем зп за год
total_salary = data_new.groupby(['club']).agg({'net_py':'sum'})
total_salary.sort_values(by='net_py',ascending = False,inplace=True)

In [75]:
total_salary.reset_index()

club     net_py
0                     PSG  240333006
1       Manchester United  163547817
2             Real Madrid  157389278
3           Bayern Munich  128184379
4                 Chelsea  115478149
5               Barcelona  114644204
6                Juventus  102262869
7         Manchester City   99485060
8               Liverpool   93089032
9             Inter Milan   85693440
10        Atletico Madrid   80651457
11                Arsenal   66353510
12                 Napoli   62602976
13              Tottenham   61627326
14                   Roma   57738878
15               AC Milan   56391978
16                Everton   53567484
17               West Ham   51385249
18             Villarreal   50567264
19              Leicester   48817187
20      Borussia Dortmund   48610450
21         Crystal Palace   48383149
22                 Monaco   45260983
23                Sevilla   44783210
24                  Lazio   42973787
25               Besiktas   40295233
26            Aston Villa   37845678
27              Newcastle   37743198
28             Fiorentina   36767875
29          Wolverhampton   36344627
30            Southampton   34903869
31                   Lyon   31842789
32                  Zenit   31832622
33          Athletic Club   30668694
34               Valencia   29032084
35              Marseille   28981256
36               Brighton   28748967
37            Galatasaray   28666136
38              Wolfsburg   28579728
39             Fenerbahce   28335766
40                Leipzig   28320516
41               Flamengo   27125009
42                 Torino   27029525
43                 Fulham   26548642
44       Bayer Leverkusen   26516174
45               Atalanta   25692784
46                Watford   25686590
47         Spartak Moscow   24869395
48                Burnley   24794403
49        Monchengladbach   24147660
50          Real Sociedad   23502165
51             Real Betis   23375100
52            Trabzonspor   22790599
53            Bournemouth   22371025
54              Krasnodar   22129850
55       Atletico Mineiro   21300010
56             Hoffenheim   21214976
57                Benfica   21108238
58             Schalke 04   20544065
59               Cagliari   20411915
60                Norwich   20224945
61                  Genoa   20000218
62              Lokomotiv   19944311
63                 Empoli   19542787
64              Palmeiras   19190004
65    Eintracht Frankfurt   19141256
66                Bologna   19100591
67                   CSKA   18216210
68                   Ajax   18028154
69                  Parma   17987487
70          Hertha Berlin   17962076
71             Celta Vigo   17824846
72          Dynamo Moscow   17707947
73              Sampdoria   17591041
74        Adana Demirspor   16813400
75                   UANL   16650005
76            Corinthians   16475003
77            River Plate   16360001
78             Basaksehir   16259391
79                Rangers   15993081
80                  Leeds   15794148
81                   Nice   15659639
82               Sassuolo   15547823
83                 Getafe   15527484
84                   Koln   14836255
85                  Porto   14760008
86                  Lille   14744758
87                Venezia   14521123
88         Karagumrukspor   14470302
89          West Bromwich   14419694
90               Espanyol   14302571
91               Hannover   14185669
92              Sao Paulo   13915018
93                America   13810001
94           Gaziantep BB   13763810
95               Bordeaux   13479181
96           Boca Juniors   13460002
97              Monterrey   13415000
98          Internacional   13295010
99                  Monza   13174226
100             Brentford   13075383
101            Stoke City   12990982
102               Udinese   12894666
103      Sheffield United   12876448
104     Nottingham Forest   12810142
105               Reading   12798086
106              Augsburg   12721870
107                Rennes

## Вывод:
По таблице можно сказать что суммы от 48569786 млн евро зарплатной ведомости - это топ клубы, после этой отметки категория сильные клубы. ПСЖ вне конкуренции - сильный отрыв по зп. Остальные топ 4 имеют примерно одинаковую зп ведомость, между остальными топ клубами разница существенная.

Наш топ 5 довольно очевидный:
* PSG	240439745
* Manchester United	163547819
* Real Madrid	157389280
* Bayern Munich	128184381
* Chelsea 116225653

## Посмотрим на таблицу более подробно
Нужно найти:

* где будут российские клубы, Аякс, Спортинг, Зальцбург, Шахтер 

* Где будет первый клуб МЛС, первый мексиканский и т.д.

In [76]:
#найдем сумму платежной ведомости клуба
total_sum_test = data_new.groupby(['club','league']).agg({'net_py':'sum'})
total_sum_test.sort_values(by='net_py',ascending = False,inplace=True)
total_sum_test.head(3)

net_py
club              league                   
PSG               Ligue 1         240333006
Manchester United Premier League  163547817
Real Madrid       La Liga         157389278

In [77]:
#найдем максимальную зп
total_sum_2 = data_new.groupby(['club','league']).agg({'net_py':'max'})
total_sum_2.sort_values(by='net_py',ascending = False,inplace=True)

In [78]:
test_two = total_sum_test.merge(total_sum_2,left_on=['club','league'],right_on=['club','league'],how='right')
display(test_two.shape)

(488, 2)

In [79]:
total = test_two.rename({'net_py_x':'net_py','net_py_y':'net_pw'}, axis=1)

In [80]:
display(total.shape)
display(total.head(4))



(488, 2)

net_py    net_pw
club              league                             
PSG               Ligue 1         240333006  50826480
Manchester United Premier League  163547817  30141507
Real Madrid       La Liga         157389278  16589763
Barcelona         La Liga         114644204  14231414

In [81]:
#сохраняем таблицу с чтобы отправить себе и потом загрузим снова с индексами
total.to_csv(r'total_ver2.csv')

In [82]:
#проставим индексы места в рейтинге
dt = pd.read_csv('total_ver2.csv',sep = ",",index_col = 0,low_memory=False)
dt = dt.replace("\'", '\"')

In [83]:
dt.groupby('club')
dt = dt.sort_values(by='net_py',ascending = False)
dt.reset_index()

club               league     net_py    net_pw
0                     PSG              Ligue 1  240333006  50826480
1       Manchester United       Premier League  163547817  30141507
2             Real Madrid              La Liga  157389278  16589763
3           Bayern Munich         1.Bundesliga  128184379  12391496
4                 Chelsea       Premier League  115478149  11212641
5               Barcelona              La Liga  114644204  14231414
6                Juventus              Serie A  102262869   8132237
7         Manchester City       Premier League   99485060  13792754
8               Liverpool       Premier League   93089032   7583603
9             Inter Milan              Serie A   85693440   7115707
10        Atletico Madrid              La Liga   80651457  10165296
11                Arsenal       Premier League   66353510   6896377
12                 Napoli              Serie A   62602976   6099178
13              Tottenham       Premier League   61627326   6896377
14                   Roma              Serie A   57738878   4574383
15               AC Milan              Serie A   56391978   7115707
16                Everton       Premier League   53567484   4135415
17               West Ham       Premier League   51385249   4135416
18             Villarreal              La Liga   50567264   5285954
19              Leicester       Premier League   48817187   4822641
20      Borussia Dortmund         1.Bundesliga   48610450   5926368
21         Crystal Palace       Premier League   48383149   4485056
22                 Monaco              Ligue 1   45260983   6902236
23                Sevilla              La Liga   44783210   6444798
24                  Lazio              Serie A   42973787   4066118
25               Besiktas            Super Lig   40295233   4726863
26            Aston Villa       Premier League   37845678   4135415
27              Newcastle       Premier League   37743198   2760962
28             Fiorentina              Serie A   36767875   3557854
29          Wolverhampton       Premier League   36344627   4509170
30            Southampton       Premier League   34903869   2592170
31                   Lyon              Ligue 1   31842789   3451118
32                  Zenit         Premier Liga   31832622   7075046
33          Athletic Club              La Liga   30668694   4055953
34               Valencia              La Liga   29032084   2978432
35              Marseille              Ligue 1   28981256   3354548
36               Brighton       Premier League   28748967   3122660
37            Galatasaray            Super Lig   28666136   3252895
38              Wolfsburg         1.Bundesliga   28579728   2968266
39             Fenerbahce            Super Lig   28335766   3374878
40                Leipzig         1.Bundesliga   28320516   4066118
41               Flamengo           Brasileiro   27125009   2680001
42                 Torino              Serie A   27029525   2033059
43                 Fulham         Championship   26548642   2242528
44       Bayer Leverkusen         1.Bundesliga   26516174   1728101
45               Atalanta              Serie A   25692784   2033059
46                Watford       Premier League   25686590   2760962
47         Spartak Moscow         Premier Liga   24869395   3049589
48                Burnley       Premier League   24794403   1892887
49        Monchengladbach         1.Bundesliga   24147660   2266861
50          Real Sociedad              La Liga   23502165   2124547
51             Real Betis              La Liga   23375100   3364713
52            Trabzonspor            Super Lig   22790599   2541324
53            Bournemouth         Championship   22371025   1892887
54              Krasnodar         Premier Liga   22129850   3039424
55       Atletico Mineiro           Brasileiro   21300010   2680000
56             Hoffenheim         1.Bundesliga   21214976   2368514
57                Benfica        Primeira Liga   21108238   2134712
58       

* первый клуб не из топ 5 это Бешикташ 26 место, следом Зенит 33 место.
* Фламенго 42 место * Спартак  - 48 место * Краснодар - 68 место * Аякс - 63 место  * Локомотив - 66 место * ЦСКА- 72 место * Динамо - 74 место * Рейнджерс - 77 место * Мексиканский UANL - 80 место * Орландо - 91 место *  Арсенал Тула - 114 место * Спортинг - 117 * Ростов - 122 * Сочи - 123 * Рубин - 135 * Ахмат - 143 * * Базель - 160 * Уфа 182 * Нижний - 206 * Урал - 217  * Крылья - 275 

## 6.Рейтинг клубов по ЗП ведомости<a name="view2"></a>

Задача: Собрать таблицу с колонками

- место в рейтинге - клуб - место в лиге по зп - размер ведомости net - максимальная зп - Перцентиль 90	- Перцентиль 80 - Перцентиль 60	- Перцентиль 40

Шаги:
1.Собрать зп ведомость по лигам и распределить клубы 
2.Собрать размезмер зп ведомости (сделано)
3.Найти максимальную зп
4.Найти перцентили и добавить в таблицу

In [84]:
percentil_40 = data_new.groupby('club')['net_py'].quantile(0.4)
percentil_60 = data_new.groupby('club')['net_py'].quantile(0.6)
percentil_80 = data_new.groupby('club')['net_py'].quantile(0.8)
percentil_90 = data_new.groupby('club')['net_py'].quantile(0.9)

In [85]:
#соберем перцентили
data_perc = pd.DataFrame()
data_perc['percentil40'] = percentil_40
data_perc['percentil60'] = percentil_60
data_perc['percentil80'] = percentil_80 
data_perc['percentil90'] = percentil_90
display(data_perc.shape)

(488, 4)

In [86]:
#максимальная зарпатная ведомость по команде
maxzp = data_new.groupby(['club','league']).agg({'net_pw':'max'})
maxzp.sort_values(by='net_pw',ascending = False,inplace=True)
display(maxzp.shape)
maxzp 

(488, 1)

net_pw
club                 league                     
PSG                  Ligue 1              977432
Manchester United    Premier League       579644
Real Madrid          La Liga              319034
Barcelona            La Liga              273681
Manchester City      Premier League       265245
Bayern Munich        1.Bundesliga         238298
Chelsea              Premier League       215628
Atletico Madrid      La Liga              195486
Arsenal              Premier League       165778
Juventus             Serie A              156389
Inter Milan          Serie A              146615
Liverpool            Premier League       145839
AC Milan             Serie A              136841
Rangers              Premiership          136796
Zenit                Premier Liga         136059
Monaco               Ligue 1              132735
Lyon                 Ligue 1              132735
Tottenham            Premier League       132623
Sevilla              La Liga              123938
Napoli               Serie A              117292
Borussia Dortmund    1.Bundesliga         113969
Aston Villa          Premier League       108509
Villarreal           La Liga              101653
Adana Demirspor      Super Lig             97743
Newcastle            Premier League        95757
Leicester            Premier League        92743
Besiktas             Super Lig             90901
Roma                 Serie A               87969
Nice                 Ligue 1               87969
Wolverhampton        Premier League        86715
Crystal Palace       Premier League        86251
Udinese              Serie A               80736
West Ham             Premier League        79527
Everton              Premier League        79527
CSKA                 Premier Liga          79172
Leipzig              1.Bundesliga          78195
Lazio                Serie A               78195
Athletic Club        La Liga               77999
Palmeiras            Brasileiro            75000
UANL                 Liga MX               72308
LA Galaxy            MLS                   69231
Fiorentina           Serie A               68420
Reading              Championship          66311
Fenerbahce           Super Lig             64902
Real Betis           La Liga               64706
Marseille            Ligue 1               64511
Getafe               La Liga               63338
Galatasaray          Super Lig             62556
America              Liga MX               61538
Brighton             Premier League        60051
Inter Miami          MLS                   58846
Hertha Berlin        1.Bundesliga          58646
Spartak Moscow       Premier Liga          58646
Porto                Primeira Liga         58646
Cagliari             Serie A               58646
Krasnodar            Premier Liga          58450
Empoli               Serie A               58255
Valencia             La Liga               57278
Wolfsburg            1.Bundesliga          57082
Sporting Lisbon      Primeira Liga         54932
Basaksehir           Super Lig             54150
Watford              Premier League        53095
Bologna              Serie A               52781
Monterrey            Liga MX               51923
LA FC                MLS                   51922
Atletico Mineiro     Brasileiro            51538
Flamengo             Brasileiro            51538
Southampton          Premier League        49849
Trabzonspor          Super Lig             48872
Venezia              Serie A               48872
Genoa                Serie A               48872
Benevento            Serie B               48872
Juventude            Brasileiro            48846
Sao Paulo            Brasileiro            48462
Schalke 04           2.Bundesliga          46721
Hoffenheim           1.Bundesliga          45548
Celta Vigo           La Liga               43984
Toronto FC           MLS                   43846
Monchengladbach      1.Bundesliga          43593
Fulham               Championship          43126
Norwich              Pre

In [87]:
#соеденим таблицы
test_2 = maxzp.merge(dt,left_on='club',right_on='club',how='right')
display(test_2.shape)
test_2 

(488, 4)

net_pw_x               league     net_py  net_pw_y
club                                                                    
PSG                     977432              Ligue 1  240333006  50826480
Manchester United       579644       Premier League  163547817  30141507
Real Madrid             319034              La Liga  157389278  16589763
Bayern Munich           238298         1.Bundesliga  128184379  12391496
Chelsea                 215628       Premier League  115478149  11212641
Barcelona               273681              La Liga  114644204  14231414
Juventus                156389              Serie A  102262869   8132237
Manchester City         265245       Premier League   99485060  13792754
Liverpool               145839       Premier League   93089032   7583603
Inter Milan             146615              Serie A   85693440   7115707
Atletico Madrid         195486              La Liga   80651457  10165296
Arsenal                 165778       Premier League   66353510   6896377
Napoli                  117292              Serie A   62602976   6099178
Tottenham               132623       Premier League   61627326   6896377
Roma                     87969              Serie A   57738878   4574383
AC Milan                136841              Serie A   56391978   7115707
Everton                  79527       Premier League   53567484   4135415
West Ham                 79527       Premier League   51385249   4135416
Villarreal              101653              La Liga   50567264   5285954
Leicester                92743       Premier League   48817187   4822641
Borussia Dortmund       113969         1.Bundesliga   48610450   5926368
Crystal Palace           86251       Premier League   48383149   4485056
Monaco                  132735              Ligue 1   45260983   6902236
Sevilla                 123938              La Liga   44783210   6444798
Lazio                    78195              Serie A   42973787   4066118
Besiktas                 90901            Super Lig   40295233   4726863
Aston Villa             108509       Premier League   37845678   4135415
Newcastle                95757       Premier League   37743198   2760962
Fiorentina               68420              Serie A   36767875   3557854
Wolverhampton            86715       Premier League   36344627   4509170
Southampton              49849       Premier League   34903869   2592170
Lyon                    132735              Ligue 1   31842789   3451118
Zenit                   136059         Premier Liga   31832622   7075046
Athletic Club            77999              La Liga   30668694   4055953
Valencia                 57278              La Liga   29032084   2978432
Marseille                64511              Ligue 1   28981256   3354548
Brighton                 60051       Premier League   28748967   3122660
Galatasaray              62556            Super Lig   28666136   3252895
Wolfsburg                57082         1.Bundesliga   28579728   2968266
Fenerbahce               64902            Super Lig   28335766   3374878
Leipzig                  78195         1.Bundesliga   28320516   4066118
Flamengo                 51538           Brasileiro   27125009   2680001
Torino                   39097              Serie A   27029525   2033059
Fulham                   43126         Championship   26548642   2242528
Bayer Leverkusen         33233         1.Bundesliga   26516174   1728101
Atalanta                 39097              Serie A   25692784   2033059
Watford                  53095       Premier League   25686590   2760962
Spartak Moscow           58646         Premier Liga   24869395   3049589
Burnley                  36402       Premier League   24794403   1892887
Monchengladbach          43593         1.Bundesliga   24147660   2266861
Real Sociedad            40857              La Liga   23502165   2124547
Real Betis               64706              La Liga   23375100   3364713
Trabzonspor              48872            Super Lig   22790599   2541324
Bournem

In [88]:
#соеденим таблицы
df_merge = test_2.merge(data_perc,left_on='club' ,right_on = 'club' )
#df_merge.sort_values(by=['net_py'],ascending=False).reset_index()
df_merge.groupby('club')
display(df_merge.shape)
df_merge 

(488, 8)

net_pw_x               league     net_py  net_pw_y  \
club                                                                       
PSG                     977432              Ligue 1  240333006  50826480   
Manchester United       579644       Premier League  163547817  30141507   
Real Madrid             319034              La Liga  157389278  16589763   
Bayern Munich           238298         1.Bundesliga  128184379  12391496   
Chelsea                 215628       Premier League  115478149  11212641   
Barcelona               273681              La Liga  114644204  14231414   
Juventus                156389              Serie A  102262869   8132237   
Manchester City         265245       Premier League   99485060  13792754   
Liverpool               145839       Premier League   93089032   7583603   
Inter Milan             146615              Serie A   85693440   7115707   
Atletico Madrid         195486              La Liga   80651457  10165296   
Arsenal                 165778       Premier League   66353510   6896377   
Napoli                  117292              Serie A   62602976   6099178   
Tottenham               132623       Premier League   61627326   6896377   
Roma                     87969              Serie A   57738878   4574383   
AC Milan                136841              Serie A   56391978   7115707   
Everton                  79527       Premier League   53567484   4135415   
West Ham                 79527       Premier League   51385249   4135416   
Villarreal              101653              La Liga   50567264   5285954   
Leicester                92743       Premier League   48817187   4822641   
Borussia Dortmund       113969         1.Bundesliga   48610450   5926368   
Crystal Palace           86251       Premier League   48383149   4485056   
Monaco                  132735              Ligue 1   45260983   6902236   
Sevilla                 123938              La Liga   44783210   6444798   
Lazio                    78195              Serie A   42973787   4066118   
Besiktas                 90901            Super Lig   40295233   4726863   
Aston Villa             108509       Premier League   37845678   4135415   
Newcastle                95757       Premier League   37743198   2760962   
Fiorentina               68420              Serie A   36767875   3557854   
Wolverhampton            86715       Premier League   36344627   4509170   
Southampton              49849       Premier League   34903869   2592170   
Lyon                    132735              Ligue 1   31842789   3451118   
Zenit                   136059         Premier Liga   31832622   7075046   
Athletic Club            77999              La Liga   30668694   4055953   
Valencia                 57278              La Liga   29032084   2978432   
Marseille                64511              Ligue 1   28981256   3354548   
Brighton                 60051       Premier League   28748967   3122660   
Galatasaray              62556            Super Lig   28666136   3252895   
Wolfsburg                57082         1.Bundesliga   28579728   2968266   
Fenerbahce               64902            Super Lig   28335766   3374878   
Leipzig                  78195         1.Bundesliga   28320516   4066118   
Flamengo                 51538           Brasileiro   27125009   2680001   
Torino                   39097              Serie A   27029525   2033059   
Fulham                   43126         Championship   26548642   2242528   
Bayer Leverkusen         33233         1.Bundesliga   26516174   1728101   
Atalanta                 39097              Serie A   25692784   2033059   
Watford                  53095       Premier League   25686590   2760962   
Spartak Moscow           58646         Premier Liga   24869395   3049589   
Burnley                  36402       Premier League   24794403   1892887   
Monchengladbach          43593         1.Bundesliga   24147660   2266861   
Real Sociedad            40857              La Liga   23502165   21245

In [89]:
df_merge = df_merge.drop(['net_pw_x'],axis=1)

In [90]:
final = df_merge
#.rename({'club':'клуб','net_pw_y':'макс зп','ligue':'лига','net_py':'размер ведомости','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
display(final.shape)
final.head(2)

(488, 7)

league     net_py  net_pw_y  percentil40  \
club                                                                  
PSG                       Ligue 1  240333006  50826480    527578.60   
Manchester United  Premier League  163547817  30141507   2963513.00   

                   percentil60  percentil80  percentil90  
club                                                      
PSG                 4159639.00   8050913.20  10673560.80  
Manchester United   4205343.60   8340758.20  11753982.20

In [91]:
#создадим фильтр лист для ранк
filter_list_rank=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]

In [92]:
dt_ranked = final.copy()
dt_ranked['rank'] = final.groupby('league')['net_py'].rank(method='dense',ascending=False)
dt_rank_filtered = dt_ranked.query("rank == @filter_list_rank")
#dt_rank_filtered.sort_values(by = ['net_py'], ascending = False)
dt_rank_filtered.head(1)

league     net_py  net_pw_y  percentil40  percentil60  percentil80  \
club                                                                        
PSG   Ligue 1  240333006  50826480    527578.60   4159639.00   8050913.20   

      percentil90  rank  
club                     
PSG   10673560.80  1.00

### 6.1 Итог таблица перцентили<a name="final1"></a>

In [93]:
dt_rank_filtered = dt_rank_filtered.rename({'club':'клуб','ligue':'лига','net_pw_y':'макс зп','net_py':'размер ведомости','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
dt_rank_filtered = dt_rank_filtered.sort_values(by = 'размер ведомости', ascending = False)
dt_rank_filtered

league  размер ведомости   макс зп  \
club                                                                    
PSG                               Ligue 1         240333006  50826480   
Manchester United          Premier League         163547817  30141507   
Real Madrid                       La Liga         157389278  16589763   
Bayern Munich                1.Bundesliga         128184379  12391496   
Chelsea                    Premier League         115478149  11212641   
Barcelona                         La Liga         114644204  14231414   
Juventus                          Serie A         102262869   8132237   
Manchester City            Premier League          99485060  13792754   
Liverpool                  Premier League          93089032   7583603   
Inter Milan                       Serie A          85693440   7115707   
Atletico Madrid                   La Liga          80651457  10165296   
Arsenal                    Premier League          66353510   6896377   
Napoli                            Serie A          62602976   6099178   
Tottenham                  Premier League          61627326   6896377   
Roma                              Serie A          57738878   4574383   
AC Milan                          Serie A          56391978   7115707   
Everton                    Premier League          53567484   4135415   
West Ham                   Premier League          51385249   4135416   
Villarreal                        La Liga          50567264   5285954   
Leicester                  Premier League          48817187   4822641   
Borussia Dortmund            1.Bundesliga          48610450   5926368   
Crystal Palace             Premier League          48383149   4485056   
Monaco                            Ligue 1          45260983   6902236   
Sevilla                           La Liga          44783210   6444798   
Lazio                             Serie A          42973787   4066118   
Besiktas                        Super Lig          40295233   4726863   
Aston Villa                Premier League          37845678   4135415   
Newcastle                  Premier League          37743198   2760962   
Fiorentina                        Serie A          36767875   3557854   
Wolverhampton              Premier League          36344627   4509170   
Southampton                Premier League          34903869   2592170   
Lyon                              Ligue 1          31842789   3451118   
Zenit                        Premier Liga          31832622   7075046   
Athletic Club                     La Liga          30668694   4055953   
Valencia                          La Liga          29032084   2978432   
Marseille                         Ligue 1          28981256   3354548   
Brighton                   Premier League          28748967   3122660   
Galatasaray                     Super Lig          28666136   3252895   
Wolfsburg                    1.Bundesliga          28579728   2968266   
Fenerbahce                      Super Lig          28335766   3374878   
Leipzig                      1.Bundesliga          28320516   4066118   
Flamengo                       Brasileiro          27125009   2680001   
Torino                            Serie A          27029525   2033059   
Fulham                       Championship          26548642   2242528   
Bayer Leverkusen             1.Bundesliga          26516174   1728101   
Atalanta                          Serie A          25692784   2033059   
Watford                    Premier League          25686590   2760962   
Spartak Moscow               Premier Liga          24869395   3049589   
Burnley                    Premier League          24794403   1892887   
Monchengladbach              1.Bundesliga          24147660   2266861   
Real Sociedad                     La Liga          23502165   2124547   
Real Betis                        La Liga          23375100   3364713   
Trabzonspor                     Super Lig          22790599   2541324   
Bournemouth         

### Теперь соберем таблицу по игрокам

In [94]:
# отсортируем лиги по зп
total_ligue = dt.groupby(['league']).agg({'net_py':'sum'})
total_ligue.sort_values(by='net_py',ascending = False,inplace=True)
total_ligue.head(2)

net_py
league                    
Premier League  1076895771
La Liga          670741782

In [95]:
filter_list=[1,3,7,12,18]

#df_new.query("rank == @filter_list")

In [96]:
data_ranked = data_new.copy()
data_ranked['rank'] = data_new.groupby('club')['net_py'].rank(method='dense',ascending=False)
data_rank_filtered = data_ranked.query("rank == @filter_list")
display(data_rank_filtered.shape)
data_rank_filtered.sort_values(by = ['club','net_py'], ascending = False)

(3020, 9)

player  net_pw   net_py position  age      country  \
1036            Ante Coric   21894  1138513        F   24      Croatia   
3129        Yanick Brecher    7624   396447        K   28  Switzerland   
3951  Nikola Boranijasevic    5865   304957        D   29       Serbia   
5396       Mirlind Kryeziu    3714   193141        D   25       Kosovo   
8106           Blaz Kramer    1759    91488        F   25     Slovenia   
...                    ...     ...      ...      ...  ...          ...   
52      Zlatan Ibrahimovic  136841  7115707        F   40       Sweden   
175         Theo Hernández   78195  4066118        D   24       France   
298           Mike Maignan   54736  2846283        K   26       France   
707            Brahim Díaz   29323  1524795        F   22        Spain   
962          Sandro Tonali   23458  1219835        M   21        Italy   

          club        league  rank  
1036    Zurich  Super League  1.00  
3129    Zurich  Super League  3.00  
3951    Zurich  Super League  7.00  
5396    Zurich  Super League 12.00  
8106    Zurich  Super League 18.00  
...        ...           ...   ...  
52    AC Milan       Serie A  1.00  
175   AC Milan       Serie A  3.00  
298   AC Milan       Serie A  7.00  
707   AC Milan       Serie A 12.00  
962   AC Milan       Serie A 18.00  

[3020 rows x 9 columns]

In [97]:
statistic_pivot = pd.pivot_table(data_rank_filtered  ,
               index=['league','club'],
               values="net_py",
               columns="rank"
               )

display(statistic_pivot.shape)

(488, 5)

In [98]:
final_player = statistic_pivot.merge(dt_rank_filtered,left_on=['league','club'],right_on=['league','club'],how='left').reset_index()
final_player = final_player[['league','club', 'rank','размер ведомости' ,1.0, 3.0, 7.0, 12.0, 18.0]]
#final_player = final_player.sort_values(by = ['net_py'], ascending = False)
final_player = final_player.fillna(0)

## 6.2.Итог таблица по игрокам<a name="final2"></a>

In [108]:
pd.options.display.max_rows = 1000
pd.set_option('display.max_colwidth', None)
pd.set_option('display.float_format', '{:.0f}'.format)

final_player_2 = final_player.rename({'league':'лига','club':'клуб','rank':'ранг','net_py':'размер ведомости',1.0:'зп 1 игрока',3.0:'зп 3 игрока',7.0:'зп 7 игрока',12.0:'зп 12 игрока',18.0:'зп 18 игрока'}, axis=1)
final_player_2 = final_player.sort_values(by = 'размер ведомости', ascending = False)
final_player_2.reset_index()

index               league                  club  rank  размер ведомости  \
0      299              Ligue 1                   PSG     1         240333006   
1      364       Premier League     Manchester United     1         163547817   
2      192              La Liga           Real Madrid     1         157389278   
3        3         1.Bundesliga         Bayern Munich     1         128184379   
4      357       Premier League               Chelsea     2         115478149   
5      180              La Liga             Barcelona     2         114644204   
6      427              Serie A              Juventus     1         102262869   
7      363       Premier League       Manchester City     3          99485060   
8      362       Premier League             Liverpool     4          93089032   
9      426              Serie A           Inter Milan     2          85693440   
10     179              La Liga       Atletico Madrid     3          80651457   
11     352       Premier League               Arsenal     5          66353510   
12     429              Serie A                Napoli     3          62602976   
13     368       Premier League             Tottenham     6          61627326   
14     430              Serie A                  Roma     4          57738878   
15     418              Serie A              AC Milan     5          56391978   
16     359       Premier League               Everton     7          53567484   
17     370       Premier League              West Ham     8          51385249   
18     196              La Liga            Villarreal     4          50567264   
19     361       Premier League             Leicester     9          48817187   
20       5         1.Bundesliga     Borussia Dortmund     2          48610450   
21     358       Premier League        Crystal Palace    10          48383149   
22     295              Ligue 1                Monaco     2          45260983   
23     194              La Liga               Sevilla     5          44783210   
24     428              Serie A                 Lazio     6          42973787   
25     473            Super Lig              Besiktas     1          40295233   
26     353       Premier League           Aston Villa    11          37845678   
27     365       Premier League             Newcastle    12          37743198   
28     423              Serie A            Fiorentina     7          36767875   
29     371       Premier League         Wolverhampton    13          36344627   
30     367       Premier League           Southampton    14          34903869   
31     292              Ligue 1                  Lyon     3          31842789   
32     387         Premier Liga                 Zenit     1          31832622   
33     178              La Liga         Athletic Club     6          30668694   
34     195              La Liga              Valencia     7          29032084   
35     293              Ligue 1             Marseille     4          28981256   
36     355       Premier League              Brighton    15          28748967   
37     476            Super Lig           Galatasaray     2          28666136   
38      17         1.Bundesliga             Wolfsburg     3          28579728   
39     475            Super Lig            Fenerbahce     3          28335766   
40      12         1.Bundesliga               Leipzig     4          28320516   
41      72           Brasileiro              Flamengo     1          27125009   
42     435              Serie A                Torino     8          27029525   
43     105         Championship                Fulham     1          26548642   
44       2         1.Bundesliga      Bayer Leverkusen     5          26516174   
45     419              Serie A              Atalanta     9          25692784   
46     369       Premier League               Watford    16          25686590   
47     384         Premier Liga        Spartak Moscow     2          24869395   
48     356       Premier League     

## 7.Рейтинг лиг по зп ведомости<a name="view3"></a>
Место в рейтинге - Лига	- Максимальная зп -  Перцентиль 90 -Перцентиль 80 -Перцентиль 60 -Перцентиль 40

In [100]:
liga = data_new.groupby(['league']).agg({'net_py':'max'})
liga.sort_values(by='net_py',ascending = False,inplace=True)
display(liga.shape)

(25, 1)

In [101]:
percent_40 = data_new.groupby('league')['net_py'].quantile(0.4)
percent_60 = data_new.groupby('league')['net_py'].quantile(0.6)
percent_80 = data_new.groupby('league')['net_py'].quantile(0.8)
percent_90 = data_new.groupby('league')['net_py'].quantile(0.9)

data_perc_ligue = pd.DataFrame()
data_perc_ligue['percentil40'] = percent_40
data_perc_ligue['percentil60'] = percent_60
data_perc_ligue['percentil80'] = percent_80 
data_perc_ligue['percentil90'] = percent_90
display(data_perc_ligue.shape)

(25, 4)

In [102]:
#соеденим таблицы 
rating_ligue = liga.merge(data_perc_ligue ,left_on='league' ,right_on = 'league' )
display(rating_ligue.shape)

(25, 5)

## 7.1 Итоговая таблица по рейтингу лиг<a name="zp1"></a>

In [103]:
#Рейтинг лиг по зп ведомостям
final_ligue = rating_ligue.rename({'league':'лига','net_py':'макс зп','percentil40':'перцентиль40','percentil60':'перцентиль60','percentil80':'перцентиль80','percentil90':'перцентиль90'}, axis=1)
final_ligue = final_ligue.sort_values(by=['макс зп'],ascending=False).reset_index()
final_ligue.groupby('league')
display(final_ligue.shape)
final_ligue



(25, 6)

league   макс зп  перцентиль40  перцентиль60  перцентиль80  \
0               Ligue 1  50826480     121984.00     274463.00     599752.00   
1        Premier League  30141507     882543.60    1724093.00    2760962.00   
2               La Liga  16589763     315124.00     630248.00    1463803.00   
3          1.Bundesliga  12391496     223637.00     453372.20     914878.40   
4               Serie A   8132237     406611.00     813223.00    1524794.00   
5          Premier Liga   7075046     304959.00     548926.00     853885.00   
6             Super Lig   5082648     172810.00     345620.00     677009.00   
7            Brasileiro   3900000      50000.00     150000.00     330001.00   
8               Liga MX   3760000     100000.00     210000.00     362000.60   
9                   MLS   3600000      79200.00     165216.00     330000.00   
10          Premiership   3556698      36170.00      48226.00      84396.00   
11         Championship   3448188     144679.00     253188.00     470208.00   
12        Primeira Liga   3049589      40661.00      60992.00     111818.00   
13              Serie B   2541324      81322.00     152479.00     280561.40   
14         2.Bundesliga   2429506     101653.00     162645.00     355785.00   
15           Eredivisie   2073720      40661.00      71157.00     162645.00   
16            La Liga 2   2033059      60992.00     101653.00     193141.00   
17     First Division A   1728100      60992.00     121984.00     203305.60   
18  Argentinian Primera   1300000     100000.00     160000.00     300000.00   
19         Super League   1189340      91488.00     142314.00     243967.80   
20           Bundesliga   1067356      40661.00      71157.00     132149.00   
21           League One    831906      48226.00      84396.00     120566.00   
22             Colombia    690000      30000.00      60000.00     100000.00   
23              Ligue 2    640414      30496.00      71157.00     121984.00   
24           League Two    168792      24113.00      36170.00      60283.00   

    перцентиль90  
0     1209670.00  
1     4135415.00  
2     3067886.00  
3     1597984.20  
4     2541324.10  
5     1242199.00  
6     1075488.00  
7      582000.00  
8      500000.00  
9      506398.80  
10     253189.00  
11     687226.00  
12     264298.00  
13     406612.30  
14     518430.00  
15     264298.00  
16     294794.00  
17     325289.00  
18     460000.90  
19     334437.50  
20     182975.00  
21     168793.00  
22     160000.00  
23     182975.00  
24      72942.80